ThePhasesOfCraftship/2_best_practice_first/testingBasics/exercises/1_Red_Green_Refactor/1_2_Palindrome/src/index.ts

```diff
@@ -0,0 +1,21 @@
+export interface IPalindromeChecker {
+  isAPalindrome: (word: string) => boolean
+  getReverseWord: (word: string) => string
+  getWordWithoutSpaces: (word: string) => string
+}
+
+export default class PalindromeChecker implements IPalindromeChecker {
+  
+  getReverseWord(word: string): string {
+    return word.split('').reverse().join('')
+  }
+  
+  getWordWithoutSpaces(word: string): string {
+    return word.replace(/\s/g, '')
+  }
+  
+  isAPalindrome(word: string) {
+    const reverseWord = this.getReverseWord(this.getWordWithoutSpaces(word)).toLowerCase()
+    return this.getWordWithoutSpaces(word).toLowerCase() == reverseWord
+  }
```

Comment:
since the class is stateless, I'd suggest making all methods static.

Reply:
I'm not sure is it ok to define methods as static and private. Because static methods are tied to the blueprint/class and private ones are meant to be used only inside the class, we're saying make the methods impossible to access outside the class and tie them to class.

Maybe I'm wrong?

Reply:
I think the two are not mutually exclusive:
```ts
export default class PalindromeChecker {
  
  private static getReverseWord(word: string): string {
    return word.split('').reverse().join('')
  }
  
  private static getWordWithoutSpaces(word: string): string {
    return word.replace(/\s/g, '')
  }
  
  static isAPalindrome(word: string) {
    // "this" here will refer to the class "blueprint" itself, not to an instance of the class
    const reverseWord = this.getReverseWord(this.getWordWithoutSpaces(word)).toLowerCase()
    return this.getWordWithoutSpaces(word).toLowerCase() == reverseWord
  }
}
```

It's just nitpicking, this is my personal preference.
Ofc in this case the class couldn't implement that interface, but it could still satisfy it:
``` ts
import { PalindromeChecker,  IPalindromeChecker } from ".";

const palindromeChecker: IPalindromeChecker = PalindromeChecker;
```

My preference was probably shaped by the [class-methods-use-this](https://eslint.org/docs/latest/rules/class-methods-use-this) eslint rule that was set up in most projects I worked with.